src/Symfony/Component/Validator/Context/ExecutionContext.php

```diff
@@ -286,7 +288,13 @@ public function getGroup()
      */
     public function getClassName()
     {
-        return $this->metadata instanceof ClassBasedInterface ? $this->metadata->getClassName() : null;
+        if ($this->metadata instanceof ClassBasedInterface) {
+            trigger_error('The Symfony\Component\Validator\ClassBasedInterface interface is deprecated since version 2.5 and will be removed in 3.0.', E_USER_DEPRECATED);
```

Comment:
this is wrong, because the behavior of having a class name is not deprecated. The new interface still provide it (it actually extends the ClassBasedInterface for BC reasons, which is why we use the parent class here).

this warning should be removed entirely. The deprecation trigger for this case is handled by the DebugClassLoader which warns you when you implement this interface in your own class.